src/routes/device-logs.ts

```diff
@@ -310,13 +318,25 @@ function handleStreamingWrite(
 			if (buffer.length && backend.available) {
 				// Even if the connection was closed, still flush the buffer
 				const promise = backend.publish(ctx, buffer);
+				// Separately publish to Loki until Redis is retired
+				let promise2;
+				try {
+					promise2 = LOKI_HOST ? loki.publish(ctx, buffer) : undefined;
+				} catch (err) {
+					captureException(err, 'Failed to create request to Loki');
+				}
 				// Clear the buffer
 				buffer.length = 0;
 				// Resume in case it was paused due to buffering
 				if (req.isPaused()) {
 					req.resume();
 				}
 				await promise;
+				try {
+					if (promise2) await promise2;
```

Comment:
I think this can be just
```suggestion
					await promise2;
```
and aside from that we always use `{}` for if's - I'll have to check if it's possible to make prettier enforce them (I thought it did tbh)

Reply:
https://github.com/balena-io-modules/node-balena-lint/pull/64 - once that's merged and updated to then it'll automatically add the braces for you so you can type it in whichever format you want and not worry :)

Reply:
Indeed `await undefined` does not throw. Thanks for the lint PR.